src/components/login/login.js

```diff
@@ -0,0 +1,61 @@
+import React, { useState } from 'react';
+import { useForm } from 'react-hook-form';
+
+import ErrorMessage from '../error-message/error-message';
+import './login.scss';
+
+function Login({ onSuccess, error }) {
+  const { register, handleSubmit, errors } = useForm();
```

Comment:
I haven't come across this hook before. What does it give us over the native html form handling?

Reply:
I've been looking at React Hook Form for a while, wanting to use it on something. Previously I've used Formik extensively, which does very similar things.

The main reason is to reduce boilerplate dealing with value setting and updating, setState onChange etc. It also has some pretty decent validation built in.